JsonException when parsing single value JSON

According to [RFC 7159](https://tools.ietf.org/html/rfc7159#section-2), "A JSON text is a sequence of tokens.  The set of tokens includes six structural characters, strings, numbers, and three literal names."
As such, TRUE, FALSE, NULL, Strings, and numbers should be acceptable JSON even if the response does not start with { or [. For example, a RESTFul response of **false** is valid JSON. 

However, RestAssured fails when parsing such single token JSON responses. Below is the exception trace:
```
groovy.json.JsonException: A JSON payload should start with an openning curly brace '{' or an openning square bracket '['.
Instead, 'false' was found on line: 1, column: 1
       at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
       at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
       at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
       at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:83)
       at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:105)
       at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:247)
       at io.restassured.internal.path.json.ConfigurableJsonSlurper.parse(ConfigurableJsonSlurper.groovy:114)
       at io.restassured.internal.path.json.ConfigurableJsonSlurper$parse.callCurrent(Unknown Source)
       at io.restassured.internal.path.json.ConfigurableJsonSlurper.parseText(ConfigurableJsonSlurper.groovy:83)
       at 
```

`
io.restassured.path.json.exception.JsonPathException: Failed to parse the JSON document
	at io.restassured.path.json.JsonPath$ExceptionCatcher.invoke(JsonPath.java:1002)
	at io.restassured.path.json.JsonPath$4.doParseWith(JsonPath.java:967)
	at io.restassured.path.json.JsonPath$JsonParser.parseWith(JsonPath.java:1047)
	at io.restassured.path.json.JsonPath.get(JsonPath.java:202)
	at files.DynamicJson.addbook(DynamicJson.java:23)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:114)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
Caused by: groovy.json.JsonException: Lexing failed on line: 1, column: 1, while reading '<', no possible valid JSON value or punctuation could be recognized.
	at groovy.json.JsonLexer.nextToken(JsonLexer.java:86)
	at groovy.json.JsonLexer$nextToken.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:130)
	at io.restassured.internal.path.json.ConfigurableJsonSlurper.parse(ConfigurableJsonSlurper.groovy:97)
	at io.restassured.internal.path.json.ConfigurableJsonSlurper$parse.callCurrent(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)
	at io.restassured.internal.path.json.ConfigurableJsonSlurper.parseText(ConfigurableJsonSlurper.groovy:83)
	at io.restassured.path.json.JsonPath$4$1.method(JsonPath.java:965)
	at io.restassured.path.json.JsonPath$ExceptionCatcher.invoke(JsonPath.java:1000)
	... 29 more
`

Any Update on this i am  facing issue of Parsing Json.

--- a/json-path/src/main/groovy/io/restassured/internal/path/json/ConfigurableJsonSlurper.groovy
+++ b/json-path/src/main/groovy/io/restassured/internal/path/json/ConfigurableJsonSlurper.groovy
@@ -105,10 +105,12 @@ class ConfigurableJsonSlurper {
       content = parseObject(lexer)
     } else if (token.getType() == OPEN_BRACKET) {
       content = parseArray(lexer)
+    } else if (token.getType().ordinal() >= NULL.ordinal()) {
+      content = token.getValue()
     } else {
       throw new JsonException(
               "A JSON payload should start with " + OPEN_CURLY.getLabel() +
-                      " or " + OPEN_BRACKET.getLabel() + ".\n" +
+                      ", " + OPEN_BRACKET.getLabel() + " or a primitive value.\n" +
                       "Instead, '" + token.getText() + "' was found " +
                       "on line: " + token.getStartLine() + ", " +
                       "column: " + token.getStartColumn()

--- a/json-path/src/test/java/io/restassured/path/json/JsonPathNumberTest.java
+++ b/json-path/src/test/java/io/restassured/path/json/JsonPathNumberTest.java
@@ -17,6 +17,7 @@
 package io.restassured.path.json;
 
 import io.restassured.path.json.config.JsonPathConfig;
+import org.junit.Ignore;
 import org.junit.Test;
 
 import java.math.BigDecimal;
@@ -105,6 +106,20 @@ public class JsonPathNumberTest {
 
         // Then
         assertThat(orderNumber, equalTo(new BigInteger(EXPECTED_LONG)));
+    }
+
+    @Ignore("Depends on GROOVY-11123")
+    @Test public void
+    json_path_returns_big_integer_for_primitive_number_when_configured_accordingly() {
+        // Given
+        final JsonPath jsonPath = new JsonPath(
+                "12345").using(new JsonPathConfig().numberReturnType(JsonPathConfig.NumberReturnType.BIG_INTEGER));
+
+        // When
+        BigInteger number = jsonPath.get("$");
+
+        // Then
+        assertThat(number, equalTo(new BigInteger("12345")));
     }
 
 }

--- a/json-path/src/test/java/io/restassured/path/json/JsonPathNumberTest.java
+++ b/json-path/src/test/java/io/restassured/path/json/JsonPathNumberTest.java
@@ -108,7 +108,6 @@ public class JsonPathNumberTest {
         assertThat(orderNumber, equalTo(new BigInteger(EXPECTED_LONG)));
     }
 
-    @Ignore("Depends on GROOVY-11123")
     @Test public void
     json_path_returns_big_integer_for_primitive_number_when_configured_accordingly() {
         // Given

--- a/json-path/src/test/java/io/restassured/path/json/JsonPathTest.java
+++ b/json-path/src/test/java/io/restassured/path/json/JsonPathTest.java
@@ -803,4 +803,51 @@ public class JsonPathTest {
         // Then   no exception should be thrown
         assertThat(jsonPath.getString("root.items[0]"), is(nullValue()));
     }
+
+    @Test public void
+    does_not_fail_on_primitive_string() {
+        // When
+        String json = "\"foo\"";
+
+        // When
+        JsonPath jsonPath = JsonPath.from(json);
+
+        // Then
+        assertThat(jsonPath.get("$"), is("foo"));
+    }
+
+    @Test public void
+    does_not_fail_on_primitive_true() {
+        // When
+        String json = "true";
+
+        // When
+        JsonPath jsonPath = JsonPath.from(json);
+
+        // Then
+        assertThat(jsonPath.get("$"), is(true));
+    }
+
+    @Test public void
+    does_not_fail_on_primitive_false() {
+        // When
+        String json = "false";
+
+        // When
+        JsonPath jsonPath = JsonPath.from(json);
+
+        // Then
+        assertThat(jsonPath.get("$"), is(false));
+    }
+    @Test public void
+    does_not_fail_on_primitive_null() {
+        // When
+        String json = "null";
+
+        // When
+        JsonPath jsonPath = JsonPath.from(json);
+
+        // Then
+        assertThat(jsonPath.get("$"), nullValue());
+    }
 }

--- a/pom.xml
+++ b/pom.xml
@@ -58,7 +58,7 @@
     <properties>
         <scm.branch>master</scm.branch>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <groovy.version>4.0.11</groovy.version>
+        <groovy.version>4.0.14</groovy.version>
         <groovy.range>[4.0,5.0)</groovy.range>
         <gmaven.version>1.5</gmaven.version>
         <hamcrest.version>2.2</hamcrest.version>

